domains/zeto/internal/zeto/handler_events.go

```diff
@@ -172,7 +173,12 @@ func (z *Zeto) handleIdentityRegisteredEvent(ctx context.Context, smtKycTree *co
 			}
 			err = z.updateMerkleTree(ctx, smtKycTree.Tree, smtKycTree.Storage, txData.TransactionID, []pldtypes.HexUint256{*pldtypes.MustParseHexUint256("0x" + publicKeyHash.Text(16))})
 			if err != nil {
-				return i18n.NewError(ctx, msgs.MsgErrorUpdateSMT, "IdentityRegistered", err)
+				// Check if this is a "key already exists" error - if so, treat it as success for idempotency
```

Comment:
This string parsing was mainly a workaround for bugs with state uniqueness. @jimthematrix do you have a thought on how we should actually handle idempotency if events happen to be replayed? I think we either have to catch an error and ignore (although I'd rather have something more explicit than string parsing), or we have to query before insert.